### DISCUSSION TITLE
What do you think about the Hack pipe operator?

### DISCUSSION THREAD

**Discussion by steida** [gcanti/fp-ts#1582] [General]
https://github.com/tc39/proposal-pipeline-operator

**Comment by samhh** [comment]
It's disappointing. I don't think it's better than what we're doing in userland now.

```ts
// Now
pipe(x, f, g)

// F#
x |> f |> g

// Hack
x |> f(^) |> g(^)
```

**Comment by DenisFrezzato** [comment]
As far as data-last curried functions are concerned, I don't see any reason to use the Hack pipe operator, I'd stick with `pipe`. I also wonder if it would even work with TypeScript's inference.

**Comment by val-o** [comment]
Oh, they actually went with ugly hack proposal... disappointing, considering that most people in discussions voted for f# one.

**Reply by val-o** [reply]
Interesting article describing why the hack style is better for javascript 

https://jamesdigioia.com/hack-pipe-for-functional-programmers-how-i-learned-to-stop-worrying-and-love-the-placeholder/

**Comment by mohaalak** [comment]
with current `fp-ts` implementation the `pipe` function is better, but if change it to something like

```
getUser()
|> Either.fold(^, notFound, identity)
```

so the data type is the first element this will make more sense but I don't know what they were thinking, I event write some comments on repository but I don't think anything changes.
